src/averageAges.js

```diff
@@ -55,7 +71,21 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const allChildren = people
+    .filter(person => people.some(mama => mama.name === person.mother));
+
+  const maleChildren = allChildren.filter(person => person.sex === 'm');
```

Comment:
You use here `filter` > `some` and then again `filter`. Let's use only `filter` > `some` without additional filtering